src/modules/dashboard/components/ActionsPage/TotalStakeWidget/TotalStakeWidget.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useEffect } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import formatNumber from 'format-number';
+import { bigNumberify } from 'ethers/utils';
+
+import Heading from '~core/Heading';
+import ProgressBar from '~core/ProgressBar';
+import { useLoggedInUser } from '~data/index';
+import {
+  useStakeAmountsForMotionQuery,
+  useColonyNativeTokenQuery,
+  useTokenInfoLazyQuery,
+} from '~data/generated';
+import { Address } from '~types/index';
+
+import styles from './TotalStakeWidget.css';
+
+const displayName = 'TotalStakeWidget';
+
+type Props = {
+  tokenDecimals: number;
+  colonyAddress: Address;
+  motionId: string;
+  isObjectionStake?: boolean;
+};
+
+const MSG = defineMessages({
+  motionTitle: {
+    id: 'dashboard.ActionsPage.TotalStakeWidget.motionTitle',
+    defaultMessage: 'Stake',
+  },
+  objectionTitle: {
+    id: 'dashboard.ActionsPage.TotalStakeWidget.objectionTitle',
+    defaultMessage: 'Goal',
+  },
+  stakeProgress: {
+    id: 'dashboard.ActionsPage.TotalStakeWidget.stakeProgress',
+    defaultMessage: '{totalPercentage}% of {requiredStake} {tokenSymbol}',
+  },
+  userStake: {
+    id: 'dashboard.ActionsPage.TotalStakeWidget.userStake',
+    defaultMessage: `You staked {userPercentage}% of this motion ({userStake} {tokenSymbol}).`,
+  },
+});
+
+const TotalStakeWidget = ({
+  tokenDecimals,
+  colonyAddress,
+  motionId,
+  isObjectionStake = false,
+}: Props) => {
+  const { walletAddress } = useLoggedInUser();
+  const { data, loading } = useStakeAmountsForMotionQuery({
+    variables: {
+      colonyAddress,
+      userAddress: walletAddress,
+      motionId,
+      isObjectionStake,
+      tokenDecimals,
+    },
+  });
+  const {
+    data: nativeTokenAddressData,
+    loading: loadingNativeTokenAddress,
+  } = useColonyNativeTokenQuery({
+    variables: { address: colonyAddress },
+  });
+  const [
+    fetchTokenInfo,
+    { data: tokenInfoData, loading: loadingTokenInfoData },
+  ] = useTokenInfoLazyQuery();
+
+  useEffect(() => {
+    if (nativeTokenAddressData) {
+      const {
+        processedColony: { nativeTokenAddress },
+      } = nativeTokenAddressData;
+      fetchTokenInfo({ variables: { address: nativeTokenAddress } });
+    }
+  }, [fetchTokenInfo, nativeTokenAddressData]);
+
+  if (loading || !data?.stakeAmountsForMotion) {
+    return null;
+  }
+
+  const { totalStaked, userStake, requiredStake } = data.stakeAmountsForMotion;
+  const divisibleRequiredStake = requiredStake !== '0' ? requiredStake : 1;
+  const totalStakedPercentage = bigNumberify(totalStaked)
+    .mul(100)
+    .div(divisibleRequiredStake)
+    .toNumber();
+  const userStakePercentage = bigNumberify(userStake)
+    .mul(100)
+    .div(divisibleRequiredStake)
+    .toNumber();
```

Comment:
So, I didn't convert these to string because they are passed to `formatNumber`, which I think only accepts numbers. 

Also, considering that they are percentages of the progress made in the staking both of their values should go from `0%` to `100%`, that means it should be safe for us to convert them to string, right? 

If the values of both these consts grow so large that they overflow, then that's more indicative that something went wrong in the background, no?

Reply:
Ok, sure, leave them as-is